strings/base_implements.h

```diff
@@ -1156,44 +1156,48 @@ namespace winrt::impl
                 return 0;
             }
 
-            if constexpr (is_agile::value)
+            return query_interface_common(id, object);
+        }
+
+        WINRT_IMPL_NOINLINE int32_t query_interface_common(guid const& id, void** object) noexcept
```

Comment:
Is this really an improvement? `query_interface_common` is still within a template (so that things like `is_agile` works) but now the guid comparison for `IAgileObject` is always performed even when it is not needed. Same for `IWeakReferenceSource` and `IInspectable`. Even if there is a slight code size improvement, this ends up being more costly at run time as those QI evaluations are costly.

Reply:
OK, reverting to the previous model (but keeping the split main-vs-common) retains the 140kb difference.